packages/compass-aggregations/src/components/saved-pipelines/saved-pipeline-confirmation-modals.tsx

```diff
@@ -1,11 +1,24 @@
 import React from 'react';
 import { ConfirmationModal } from '@mongodb-js/compass-components';
+import { useTrackOnChange } from '@mongodb-js/compass-logging';
 
 export const OpenPipelineConfirmationModal: React.FunctionComponent<{
   isOpen: boolean;
   onCancel: () => void;
   onConfirm: () => void;
 }> = ({ isOpen, onCancel, onConfirm }) => {
+  useTrackOnChange(
+    'COMPASS-AGGREGATIONS-UI',
+    isOpen,
+    (track) => {
+      if (isOpen) {
+        track('Srceen', { name: 'restore_pipeline_modal' });
```

Comment:
```suggestion
        track('Screen', { name: 'restore_pipeline_modal' });
```

Reply:
Ohhh, thank you for catching this, I re-read it multiple times and still missed it! I am wondering if it's maybe worth making first argument a union instead, probably a bit of an overkill, but we do know the full list, adding new ones is not much work, and feels like making typos is too easy otherwise (at least for me 🙈)

Reply:
Hm yeah, we could probably have a separate TS definition for all tracking events & their properties? I think that would be okay

Reply:
Will do that as a follow-up